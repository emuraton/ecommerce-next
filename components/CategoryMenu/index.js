import React from 'react';
import styled from 'styled-components';

import Category from '../Category';

const Container = styled.div`
  padding-left: 15px;
  padding-right: 15px;
`;

const List = styled.ul`
  padding-left: 0px;
  list-style: none;
`;

const Item = styled.li`
  display: inline-block;
  margin-right: 20px;
`;

export default class extends React.Component {
  state = {
    activeSlug: 'ticket',
  };

  onClick = e => {
    e.preventDefault();
    const activeSlug = { activeSlug: e.currentTarget.name };
    this.setState(() => activeSlug);
  };

  render() {
    const { categories } = this.props;
    const { activeSlug } = this.state;
    return (
      <Container>
        <List>
          {categories &&
            categories.map(({ pk, name, slug }) => (
              <Item key={pk}>
                <Category
                  name={name}
                  slug={slug}
                  active={slug === activeSlug}
                  onClick={this.onClick}
                />
              </Item>
            ))}
        </List>
      </Container>
    );
  }
}
