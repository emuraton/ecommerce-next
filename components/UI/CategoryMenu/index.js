// @flow
import * as React from 'react';
import styled from 'styled-components';

import Category from '../Category';

const Container = styled.div`
  width: 100%;
  position: sticky;
  top: 0px;
  display: inline-block;
  background-color: white;
  z-index: 10;
`;

const List = styled.ul`
  display: flex;
  justify-content: center;
  padding-left: 0px;
  margin-bottom: 10px;
  margin-top: 10px;
`;

const Item = styled.li`
  display: inline-block;
  margin-right: 20px;
`;

type Props = {
  categories: Array<{ pk: string, name: string, slug: string }>,
  slideIndex: number,
  onClick: (event: SyntheticMouseEvent<any>) => {},
};

const CategoryMenu = ({ categories, slideIndex, onClick }: Props) => {
  if (!categories) return null;
  return (
    <Container>
      <List>
        {categories.map(({ pk, name, slug }, index) => (
          <Item key={pk}>
            <Category name={name} slug={slug} active={index === slideIndex} onClick={onClick} />
          </Item>
        ))}
      </List>
    </Container>
  );
};

export default CategoryMenu;
