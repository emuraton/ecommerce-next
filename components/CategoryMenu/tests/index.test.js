import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CategoryMenu from '../';

describe('<CategoryMenu />', () => {
  const initialProps = {
    categories: [
      { pk: 1, slug: 'ticket', name: 'Ticket', products: [] },
      { pk: 2, slug: 'hotel', name: 'Hotel', products: [] },
    ],
  };
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<CategoryMenu {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('@event handlers', () => {
    it('onClick', () => {
      const mockEvent = {
        currentTarget: { name: 'hotel' },
        preventDefault: Function.prototype,
      };
      const wrapper = shallow(<CategoryMenu {...initialProps} />);

      expect(wrapper.state('activeSlug')).toBe('ticket');
      wrapper.instance().onClick(mockEvent);
      expect(wrapper.state('activeSlug')).toBe('hotel');
    });
  });
});
