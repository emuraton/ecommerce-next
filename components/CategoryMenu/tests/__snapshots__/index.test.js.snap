// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryMenu /> @renders default render 1`] = `
<CategoryMenu__Container>
  <CategoryMenu__List>
    <CategoryMenu__Item
      key="1"
    >
      <Category
        active={true}
        name="Ticket"
        onClick={[Function]}
        slug="ticket"
      />
    </CategoryMenu__Item>
    <CategoryMenu__Item
      key="2"
    >
      <Category
        active={false}
        name="Hotel"
        onClick={[Function]}
        slug="hotel"
      />
    </CategoryMenu__Item>
  </CategoryMenu__List>
</CategoryMenu__Container>
`;
